### PR TITLE
Azure Pipelines : Fix macOS ParaView/Python 3.9 compatibility

### PR DESCRIPTION
**Pull request by pierre-guillou**
Once finished, this PR should fix the Azure Pipelines CI on macOS.

Enjoy,
Pierre

### CODE REVIEW

**Comment by pierre-guillou** [comment]
So, after several tries, I stumbled upon an issue preventing the caching of the ParaView install folder. This issue is twofold:
* VTK from ParaView 5.8 uses a hard-coded path to the Python headers directory in the generated VTK-targets.cmake file (target `VTK::Python`, property `INTERFACE_INCLUDE_DIRECTORIES`). This does not seems to be the case with ParaView 5.9.
* Homebrew Python headers are not located in a fixed directory: sometimes, a parent directory might be renamed (randomly? depending on the virtual machine?).

Those two issues prevent the use of the Pipeline cache system to cache the ParaView installation folder because if the hard-coded path to the Python headers in VTK and their real path in the system mismatch, TTK fails to configure.

For now, I have removed this cache task. We still have the secondary cache with ccache that can fasten the whole build time, sometimes as fast as using the installation folder cache.

To sum up, this PR
* adds the support of Python 3.9 to ParaView 5.8 through the TTK patches
* reworks the macOS Azure CI to build ParaView 5.8 from scratch before TTK (in the same VM) in order to access the same (Homebrew) Python
* uses Ninja instead of Make in the macOS Azure CI
* enable the ttkHelloWorldCmd standalone test in the macOS Azure CI

Hope this will work fine this time.
Enjoy,
Pierre

**Comment by julien-tierny** [comment]
alright, looks good. thanks!
